doc/CONFIGURATION.md

```diff
@@ -517,6 +518,28 @@ const node = await Libp2p.create({
 })
 ```
 
+#### Configuring Transport Manager
+
+The Transport Manager is responsible for managing the libp2p transports life cycle. This includes starting listeners for the provided listen addresses, closing these listeners and dialing using the provided transports. By default, if a libp2p node has a list of multiaddrs for listenning on and there are no valid transports for those multiaddrs, libp2p will throw an error on startup and shutdown. However, for some applications it is perfectly acceptable for libp2p nodes to start in dial only mode if all the listen multiaddrs failed. This error tolerance can be enabled as follows:
+
+```js
+const Libp2p = require('libp2p')
+const TCP = require('libp2p-tcp')
+const MPLEX = require('libp2p-mplex')
+const SECIO = require('libp2p-secio')
+
+const node = await Libp2p.create({
+  modules: {
+    transport: [TCP],
+    streamMuxer: [MPLEX],
+    connEncryption: [SECIO]
+  },
+  transportManager: {
+    supportDialOnly: true
```

Comment:
We originally talked about this being a bit more flexible, https://github.com/libp2p/js-libp2p-websocket-star/issues/61#issuecomment-437425776. Perhaps we could change this to something like `faultTolerance`, and we could export an enum of values that could be used. For now we could just support `FATAL_ALL` (no transport was able to listen) and `NO_FATAL` (no listening is required).

Reply:
I didn't see this discussion. I agree with adding more flexibility :)
Should we go with `NO_FATAL` as the default?

Reply:
`FATAL_ALL`, as that's the current behavior. In reality this should be different for browsers/node, but we can update that when we add base configuration support.